store/sdk/src/main/java/org/apache/carbondata/sdk/file/CSVCarbonWriter.java

```diff
@@ -65,7 +65,7 @@
   @Override
   public void write(Object object) throws IOException {
     try {
-      writable.set((String[]) object);
+      writable.set((Object[]) object);
```

Comment:
better to keep as string, as converter step expecting binary as string ?

Reply:
It can support write with real data type.

`
          String labelFileName = ((String) files[i]).split(sufImage)[0] + sufAnnotation;
          BufferedInputStream txtBis = new BufferedInputStream(new FileInputStream(labelFileName));
          String labelValue = null;
          byte[] labelBinary = null;
          labelBinary = new byte[txtBis.available()];
          while ((txtBis.read(labelBinary)) != -1) {
            labelValue = new String(labelBinary, "UTF-8");
          }
          // write data
          writer.write(new Object[]{i, (String) files[i], originBinary,
              labelFileName, labelValue});`

Reply:
but CSVWriter is always going for convert step. so if you send as byte[] instead of string, decode hex may give wrong result ?

Reply:
no need convert byte[] to string before write, and no need convert string to byte[] in carbon internal . No encode hex for byte[]